routing/base/astar_algorithm.hpp

```diff
@@ -13,18 +13,43 @@
 namespace routing
 {
 
-template <typename TVertexType>
+template <typename TWeightType>
+constexpr TWeightType GetWeightZero();
+template <>
+constexpr double GetWeightZero<double>()
+{
+  return 0.0;
+}
+
+// Precision of comparison weights.
+template <typename TWeightType>
+constexpr TWeightType GetWeightEpsilon();
+template <>
+constexpr double GetWeightEpsilon<double>()
+{
+  return 1e-6;
+}
+
+template <typename TWeightType>
+constexpr TWeightType GetWeightMax();
+template <>
+constexpr double GetWeightMax<double>()
```

Comment:
it'll be
template <typename WeightType>
struct WeightLimits
{
  WeightType constexpr kEpsilon = ...;
  WeightType constexpr kMax = ...;
  ...
}
cause we can not use DoubleWeight from AStar (we do not know weight type).